src/resolve/resolved-manager.c

```diff
@@ -1929,3 +1930,99 @@ void dns_manager_reset_statistics(Manager *m) {
+        if (r < 0)
+                return r;
+
+        /* Build the global configuration first. */
+        LIST_FOREACH(servers, s, m->dns_servers) {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+
+                r = sd_json_variant_new_string(&v, dns_server_string_full(s));
+                if (r < 0)
+                        return r;
+
+                r = sd_json_variant_append_array(&global_dns, v);
+                if (r < 0)
+                        return r;
+        }
+
+        LIST_FOREACH(domains, d, m->search_domains) {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+
+                r = sd_json_variant_new_string(&v, DNS_SEARCH_DOMAIN_NAME(d));
+                if (r < 0)
+                        return r;
+
+                r = sd_json_variant_append_array(&global_search_domains, v);
+                if (r < 0)
+                        return r;
+        }
+
+        if (m->current_dns_server)
+                current_dns_server = dns_server_string_full(m->current_dns_server);
+
+        r = sd_json_buildo(
+                        &global,
+                        SD_JSON_BUILD_PAIR_STRING("interface", NULL),
+                        SD_JSON_BUILD_PAIR_UNSIGNED("interfaceIndex", 0),
+                        SD_JSON_BUILD_PAIR_STRING("currentDNSServer", current_dns_server),
+                        SD_JSON_BUILD_PAIR_VARIANT("dnsServers", global_dns),
+                        SD_JSON_BUILD_PAIR_VARIANT("searchDomains", global_dns));
+        if (r < 0)
+                return r;
+
+        r = sd_json_variant_append_array(&configuration, global);
+        if (r < 0)
+                return r;
+
+        /* Append configuration for each link */
+        HASHMAP_FOREACH(l, m->links) {
+                _cleanup_(sd_json_variant_unrefp) sd_json_variant *link_configuration = NULL;
+
+                r = link_dump_configuration_json(l, &link_configuration);
+                if (r < 0)
+                        return r;
+
+                r = sd_json_variant_append_array(&configuration, link_configuration);
+                if (r < 0)
+                        return r;
+        }
+
+        return sd_json_buildo(ret, SD_JSON_BUILD_PAIR_VARIANT("configuration", configuration));
+}
+
+int manager_send_dns_configuration_changed(Manager *m) {
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *configuration = NULL;
+        int r;
+
+        assert(m);
+
+        if (set_isempty(m->varlink_dns_configuration_subscription))
+                return 0;
+
+        r = manager_dump_dns_configuration_json(m, &configuration);
+        if (r < 0)
+                return log_error_errno(r, "Failed to dump DNS configuration json: %m");
+
+        r = varlink_many_notify(m->varlink_dns_configuration_subscription, configuration);
+        if (r < 0)
+                log_error_errno(r, "Failed to send DNS configuration event, ignoring: %m");
```

Comment:
Please downgrade the log level to warning.
Also, please propagate the error, even though it is ignored by the caller.